image-classifier/index.ts

```diff
@@ -0,0 +1,165 @@
+  const classificationResult = await imageClassifier.classify(event.target);
+  // Write the predictions to a new paragraph element and add it to the DOM.
+  const classifications = classificationResult.classifications;
+
+  const p = event.target.parentNode.childNodes[3];
+  p.className = 'classification';
+  p.innerText = `Classification: ${classifications[0].categories[0].categoryName}
+                 Confidence: ${Math.round(parseFloat(`${classifications[0].categories[0].score}`) * 100)}%`;
+    // 'Classificaton: ' +
+    // classifications[0].categories[0].categoryName +
+    // '\n Confidence: ' +
+    // Math.round(parseFloat(`${classifications[0].categories[0].score}`) * 100) +
+    // '%';
+}
+
+/********************************************************************
+// Demo 2: Continuously grab image from webcam stream and classify it.
+********************************************************************/
+
+// Check if webcam access is supported.
+function hasGetUserMedia() {
+  return !!(navigator.mediaDevices && navigator.mediaDevices.getUserMedia);
+}
+
+// Get classification from the webcam
+async function predictWebcam() {
+  // Do not classify if imageClassifier hasn't loaded
+  if (imageClassifier === undefined) {
+    return;
+  }
+  // if image mode is initialized, create a new classifier with video runningMode
+  if (runningMode === 'image') {
+    runningMode = 'video';
+    // await runDemo();
+    await imageClassifier.setOptions({ runningMode: runningMode });
+  }
+  let nowInMs = Date.now();
+  // Start classifying the stream.
+  const classificationResult = await imageClassifier.classifyForVideo(
+    video,
+    nowInMs
+  );
+  video.style.height = videoHeight;
+  video.style.width = videoWidth;
+  webcamPredictions.style.width = videoWidth;
+
+  const classifications = classificationResult.classifications;
+  webcamPredictions.className = 'webcamPredictions';
+  webcamPredictions.innerText = `Classification: ${classifications[0].categories[0].categoryName}
+                                 Confidence: ${Math.round(parseFloat(`${classifications[0].categories[0].score}`) * 100)}%`;
+    // 'Classification: ' +
+    // classifications[0].categories[0].categoryName +
+    // '\n Confidence: ' +
+    // Math.round(parseFloat(classifications[0].categories[0].score) * 100) +
+    // '%';
+
+  // Call this function again to keep predicting when the browser is ready.
+  if (webcamRunning === true) {
+    window.requestAnimationFrame(predictWebcam);
+  }
+}
+
+// Enable the live webcam view and start classification.
+async function enableCam(event) {
+  if (imageClassifier === undefined) {
+    return;
+  }
+
+  if (webcamRunning === true) {
+    webcamRunning = false;
+    enableWebcamButton.innerText = 'ENABLE PREDICTIONS';
+  } else {
+    console.log('webcam was off');
+    webcamRunning = true;
+    enableWebcamButton.innerText = 'DISABLE PREDICITONS';
+  }
+
+  // getUsermedia parameters.
+  const constraints = {
```

Comment:
Consider adding `audio:false`, since we have no need for audio in image classification (and this can also relax certain browser restrictions on video permissions).

Reply:
@tyrmullen Thanks for your suggestion, I added!